css/src/components/form/checkbox.scss

```diff
@@ -116,4 +140,40 @@ $checkbox-spacing: $spacer-3 !default;
 			border-color: $checkbox-check-color;
 		}
 	}
+
+	&.is-invalid input[type='checkbox']:not(:checked) {
+		+ .checkbox-check {
+			border-color: $checkbox-danger-border-color;
+		}
+
+		@include focus-visible() {
+			+ .checkbox-check {
+				box-shadow: $checkbox-focus-box-shadow-size $checkbox-danger-border-color;
+			}
+		}
+	}
+
+	&.is-valid input[type='checkbox']:not(:checked) {
```

Comment:
I'm wondering if `is-valid` is necessary for checkbox or other form fields. 

For example, when an empty required input in a form is no longer empty, we revert the `is-invalid` color and remove any related messaging rather than making the input green.

Reply:
I added it for consistency with other form components which have both `success` and `danger` states.